File: stock-api-service/src/main/java/com/yipeng/stockapi/model/StockData.java
package com.yipeng.stockapi.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class StockData {
    private String symbol;
    private BigDecimal price;
    private BigDecimal change;
    private BigDecimal changePercent;
    private BigDecimal open;
    private BigDecimal high;
    private BigDecimal low;
    private BigDecimal previousClose;
    private Long volume;
    
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime timestamp;
    
    public StockData(String symbol, BigDecimal price) {
        this.symbol = symbol;
        this.price = price;
        this.timestamp = LocalDateTime.now();
    }
} 

File: data-processor/src/main/scala/com/yipeng/recorder/dataprocessor/model/StockData.scala
package com.yipeng.recorder.dataprocessor.model

import com.fasterxml.jackson.annotation.JsonProperty
import java.math.BigDecimal
import java.time.LocalDateTime

/**
 * Local StockData model for the data-processor module
 * This removes dependency on the stock-api-service model
 */
case class StockData(
  @JsonProperty("symbol") symbol: String,
  @JsonProperty("price") price: BigDecimal,
  @JsonProperty("change") change: Option[BigDecimal],
  @JsonProperty("changePercent") changePercent: Option[BigDecimal],
  @JsonProperty("open") open: Option[BigDecimal],
  @JsonProperty("high") high: Option[BigDecimal],
  @JsonProperty("low") low: Option[BigDecimal],
  @JsonProperty("previousClose") previousClose: Option[BigDecimal],
  @JsonProperty("volume") volume: Option[Long],
  @JsonProperty("timestamp") timestamp: LocalDateTime
) 